text/0000-decorator-support.md

```diff
@@ -0,0 +1,142 @@
+classic class model is not available. However, creating an API compatible
+classic class version using Ember's component manager API should be possible if
+users want to write Glimmer-like component's with classic syntax.
+
+### Decorator Versioning and Support
+
+Because the decorators spec has changed while remaining in a stage 2, it's no
+longer enough to refer to "stage 2" as a version. Since Babel will be providing
+the transforms, we will use their versioning system. [This is currently in
+development](https://github.com/babel/babel/pull/9360), but looks like it will
+be based on the TC39 meeting that the spec was presented at. For instance, the
+current transforms available in Babel are referred to as `nov-2018`.
+
+Ember will begin by supporting the _latest_ version of the decorators transform
+provided by Babel. Currently this is the `nov-2018` version, but by the time
+decorators are released this may change.
+
+The decorators version will be configurable on a _per-app/engine/addon_ basis.
+This will allow the ecosystem to adopt decorators without fears that if an
+application updates, it could break their decorators. If no version is
+specified, the transform will default to the latest version supported by Ember
+at that time. In many cases the only decorators in use will be Ember's, so this
+allows the ecosystem to upgrade without additional maintenance costs. The exact
+method for specifying the version will be determined during implementation, but
+will likely be a configuration option or addon.
+
+When new transforms are released, Ember will continue to support older versions
+of the transforms alongside the new ones until _at least_ the next LTS release,
+with deprecation warnings for any users of the older transforms. This will mean
+shipping multiple different versions of decorators at once, and providing a way
+for apps/addons to import the correct version. Additionally, Ember will attempt
+to provide addon authors with tools that allow _them_ to do this as well, so
+they can support multiple versions of transforms easily and transparently. The
+exact API for these tools is out of scope for this RFC, and will likely be
+dependent on their implementation.
+
+## How we teach this
+
+As mentioned in the motivation section, the _invocation_ side of decorators
+seems to be much more stable currently, so we ideally will not have to teach end
+consumers too much about the details of this system. We should definitely
+provide a disclaimer that outlines how our support structure works, and how
+users can lock their version if they are using custom decorators. This should be
+included _early_ in the guides.
+
+We should also provide thorough documentation for any tooling provided to help
+addon authors write their own decorators. This will be a bit more of a moving
+target, since this RFC does not define exact APIs and they will not be a
+priority until Ember itself prepares to support multiple versions of the
+transforms.
```

Comment:
I think it depends on a few things:

1. The complexity of the APIs
2. The number of users in the ecosystem who will have to adopt them

If there are relatively few addons who end up writing their own decorators, and the APIs are relatively simple and folks are aligned on the solution, the RFC process may not make sense here, since the APIs would be temporary.